optuna/samplers/_cmaes.py

```diff
@@ -385,38 +384,12 @@ def sample_relative(
             search_space, transform_step=not self._with_margin, transform_0_1=True
         )
 
-        if self._initial_popsize is None:
-            self._initial_popsize = 4 + math.floor(3 * math.log(len(trans.bounds)))
-
-        popsize: int = self._initial_popsize
-        n_restarts: int = 0
-        n_restarts_with_large: int = 0
-        poptype: str = "small"
-        small_n_eval: int = 0
-        large_n_eval: int = 0
-        if len(completed_trials) != 0:
-            latest_trial = completed_trials[-1]
-
-            popsize_attr_key = self._attr_keys.popsize()
-            if popsize_attr_key in latest_trial.system_attrs:
-                popsize = latest_trial.system_attrs[popsize_attr_key]
-            else:
-                popsize = self._initial_popsize
-
-            n_restarts_attr_key = self._attr_keys.n_restarts()
-            n_restarts = latest_trial.system_attrs.get(n_restarts_attr_key, 0)
-            n_restarts_with_large = latest_trial.system_attrs.get(
-                self._attr_keys.n_restarts_with_large, 0
-            )
-            poptype = latest_trial.system_attrs.get(self._attr_keys.poptype, "small")
-            small_n_eval = latest_trial.system_attrs.get(self._attr_keys.small_n_eval, 0)
-            large_n_eval = latest_trial.system_attrs.get(self._attr_keys.large_n_eval, 0)
+        if self._popsize is None:
+            self._popsize = 4 + math.floor(3 * math.log(len(trans.bounds)))
 
-        optimizer = self._restore_optimizer(completed_trials, n_restarts)
+        optimizer = self._restore_optimizer(completed_trials)
         if optimizer is None:
-            optimizer = self._init_optimizer(
-                trans, study.direction, population_size=self._initial_popsize
-            )
+            optimizer = self._init_optimizer(trans, study.direction, population_size=self._popsize)
```

Comment:
I think we could simply removing `population_size` argument from `_init_optimizer()` method. What do you think?

```suggestion
            optimizer = self._init_optimizer(trans, study.direction)
```